components/raftstore/src/store/worker/pd.rs

```diff
@@ -1464,6 +1464,16 @@ where
             }
 
             Task::Heartbeat(hb_task) => {
+                // HACK! In order to keep the compatible of protos, we use 0 to identify
+                // the size uninitialized regions, and use 1 to identify the empty regions.
+                //
+                // See tikv/tikv#11114 for details.
+                let approximate_size = match hb_task.approximate_size {
+                    Some(0) => 1,
+                    Some(v) => v,
+                    None => 0, // size uninitialized
+                };
+                let approximate_keys = hb_task.approximate_keys.unwrap_or_default();
```

Comment:
Is it necessary to do the similar hack for keys?

Reply:
the approximate keys is used in:

- /regions/check/hist-keys
- merge checker
- region stats

It seems that this sudden change in value will not affect scheduling, so there is no need to hack this value.